web/packages/teleport/src/services/ping/makePing.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PingResponse } from './types';
+
+export default function makePing(json: any): PingResponse {
+  json = json || {};
+  const { automatic_upgrades = {} } = json;
+
+  return {
+    automaticUpgrades: automatic_upgrades === true,
```

Comment:
`automatic_upgrades` may not be present in the backend response
If we remove the `=== true`, then `automaticUpgrades` resolves to `undefined` which then means extra checks when we need to use it

Reply:
That's fine, as you'll be doing either ```if (pingResponse.automaticUpgrades) {}
// or
if (!pingResponse.automaticUpgrades) {}```

Which evaluates it to truthy/falsey values

Reply:
> Edit - can we make automatic_upgrades always available in the response?

We already have `automaticUpgrades`. Should I remove it in favor of `automatic_upgrades` or keep both?
It seems you are suggesting both, why would we need both?

Reply:
As I understood it, `automatic_upgrades` is from the backend that we're converting to `automaticUpgrades` for the frontend

Regardless of what it is, it would be good to make sure the field is always in the backend response and set to `true` or` false`

Reply:
Got it :+1:

Reply:
After last [commit](https://github.com/gravitational/teleport/pull/23395/commits/5b8bc3bec13793f982656af88ab4ca9f58167f3e):

```
dinis@lenix ~> curl --silent -k https://127.0.0.1.nip.io:3080/webapi/ping | jq '.automatic_upgrades'
false
dinis@lenix ~> curl --silent -k https://127.0.0.1.nip.io:3080/webapi/ping | jq '.automatic_upgrades'
true
```